syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTPaymentWorkflowEnums";
option optimize_for = LITE_RUNTIME;

enum PaymentWorkflowType {
    PAYMENTWORKFLOWTYPE_ERROR = 0;
    PAYMENTWORKFLOWTYPE_OUTGOINGCHEQUE = 1;
    PAYMENTWORKFLOWTYPE_INCOMINGCHEQUE = 2;
    PAYMENTWORKFLOWTYPE_OUTGOINGINVOICE = 3;
    PAYMENTWORKFLOWTYPE_INCOMINGINVOICE = 4;
}

enum PaymentWorkflowState {
    PAYMENTWORKFLOWSTATE_ERROR = 0;
    PAYMENTWORKFLOWSTATE_UNSENT = 1;
    PAYMENTWORKFLOWSTATE_CONVEYED = 2;
    PAYMENTWORKFLOWSTATE_CANCELLED = 3;
    PAYMENTWORKFLOWSTATE_ACCEPTED = 4;
    PAYMENTWORKFLOWSTATE_COMPLETED = 5;
    PAYMENTWORKFLOWSTATE_EXPIRED = 6;
}

enum EventTransportMethod {
    TRANSPORTMETHOD_ERROR = 0;
    TRANSPORTMETHOD_NONE = 1;
    TRANSPORTMETHOD_OT = 2;
    TRANSPORTMETHOD_OOB = 3;
}

enum PaymentEventType {
    PAYMENTEVENTTYPE_ERROR = 0;
    PAYMENTEVENTTYPE_CREATE = 1;
    PAYMENTEVENTTYPE_CONVEY = 2;
    PAYMENTEVENTTYPE_CANCEL = 3;
    PAYMENTEVENTTYPE_ACCEPT = 4;
    PAYMENTEVENTTYPE_COMPLETE = 5;
}
